plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/PostgreUtils.java

```diff
@@ -146,14 +146,18 @@ public static boolean supportsTypeCategory(JDBCDataSource dataSource) {
 
     @Nullable
     public static <OWNER extends DBSObject, OBJECT extends PostgreObject> OBJECT getObjectById(
-            @NotNull DBRProgressMonitor monitor,
+            @Nullable DBRProgressMonitor monitor,
             @NotNull AbstractObjectCache<OWNER, OBJECT> cache,
             @NotNull OWNER owner,
             long objectId)
             throws DBException {
-        for (OBJECT object : cache.getAllObjects(monitor, owner)) {
-            if (object.getObjectId() == objectId) {
-                return object;
+        if (monitor == null) {
+            log.debug("Can't load object by ID with nullable monitor.");
```

Comment:
I think it's better to fix the case when null is passed here, rather then hiding an error, because it shouldn't happen.

Reply:
No, because the monitor is going here as a null intentionally.